packages/zero-cache/src/services/view-syncer/cvr.ts

```diff
@@ -936,29 +944,36 @@ export function getInactiveQueries(cvr: CVR): {
       continue;
     }
     for (const clientState of Object.values(query.clientState)) {
+      // 1. Take the longest TTL
+      // 2. If the query is not inactivated (for any client), do not return it
       const {inactivatedAt, ttl} = clientState;
-      if (inactivatedAt !== undefined) {
-        const clampedTTL = clampTTL(ttl);
-        const existing = inactive.get(queryID);
+      const existing = inactive.get(queryID);
+      if (inactivatedAt === undefined) {
         if (existing) {
-          // The stored one might be too large because from a previous version of
-          // zero
-          const existingTTL = clampTTL(existing.ttl);
-          // Use the last eviction time.
-          if (
-            existingTTL + ttlClockAsNumber(existing.inactivatedAt) <
-            ttlClockAsNumber(inactivatedAt) + clampedTTL
-          ) {
-            existing.ttl = clampedTTL;
-            existing.inactivatedAt = inactivatedAt;
-          }
-        } else {
-          inactive.set(queryID, {
-            hash: queryID,
-            inactivatedAt,
-            ttl: clampedTTL,
-          });
+          inactive.delete(queryID);
         }
+        break;
+      }
+
+      const clampedTTL = clampTTL(ttl);
+      if (existing) {
+        // The stored one might be too large because from a previous version of
+        // zero
+        const existingTTL = clampTTL(existing.ttl);
+        // Use the last eviction time.
```

Comment:
Would 
// Use the expiration furthest in the future.

be a more accurate comment?